File: pset6/readability/readability.py
from cs50 import get_string
import re


def main():
    # Promt user for text
    text = get_string("Text: ")
    letter = 0
    word = 1
    sentence = 0

    # Scan every character from text
    for c in text:
        # Count letter(s) from text
        if re.match("[a-zA-Z]", c):
            letter += 1
        # Count word(s) from text
        elif c == " ":
            word += 1
        # Count sentence(s) from text
        elif re.match("[.!?]", c):
            sentence += 1

    # Compute avg letter(s) per 100 words
    L = round((letter/word) * 100, 2)
    # Compute avg sentence(s) per 100 words
    S = round((sentence/word) * 100, 2)
    # Compute Coleman-Liau index
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # If index is less than 1
    if index < 1:
        print("Before Grade 1")
    # If index is 16 or greater
    elif index >= 16:
        print("Grade 16+")
    # If not of the above
    else:
        print(f"Grade {index}")


main()

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Promt user for text
    string text = get_string("Text: ");
    float letter = 0;
    float word = 1;
    float sentence = 0;
    int index = 0;

    for (int i = 0, length = strlen(text); i < length; ++i)
    {
        // Count how many letter(s) in text
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            ++letter;
        }
        // Count how many word(s) in text
        if (text[i] == 32)
        {
            ++word;
        }
        // Count how many sentence(s) in text
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            ++sentence;
        }
    }

    // Average letter(s) per 100 words
    letter = (letter / word) * 100;

    // Average sentence(s) per 100 words
    sentence = (sentence / word) * 100;

    // Compute Coleman-Liau index
    index = round((0.0588 * letter) - (0.296 * sentence) - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}